apps/condo/domains/billing/components/BillingPageContent/ReceiptsTable.tsx

```diff
@@ -49,32 +38,36 @@ export const ReceiptsTable: React.FC<IContextProps> = ({ context }) => {
     const contextPeriod = get(context, ['lastReport', 'period'], null)
     const currencyCode = get(context, ['integration', 'currencyCode'], 'RUB')
 
-    const queryMetas: Array<QueryMeta<BillingReceiptWhereInput>> = [
-        ...staticQueryMetas,
-        { keyword: 'period', filters: [periodFilter], defaultValue: contextPeriod },
-        { keyword: 'search', filters: [addressFilter, unitNameFilter, accountFilter, fullNameFilter], combineType: 'OR' },
-    ]
-    const { filtersToWhere, sortersToSortBy } = useQueryMappers(queryMetas, SORTABLE_PROPERTIES)
+    const filterMetas = useReceiptTableFilters(context)
+    const [search, handleSearchChange] = useSearch()
+    const { filtersToWhere, sortersToSortBy } = useQueryMappers(filterMetas, SORTABLE_PROPERTIES)
+    const categorySearch = categoryToSearchQuery(search, intl.messages)
+    const filtersToPass = filtersToWhere(filters)
+
+    if ( filtersToPass.AND.length > 1 ){
+        filtersToPass.AND[1].OR ? categorySearch && filtersToPass.AND[1].OR.push(categorySearch) :
+            categorySearch ? filtersToPass.AND[1].OR = categorySearch : null
```

Comment:
That looks a little strange. Could you please provide comment with description "why filters should contain more than one element".  Anyway it's too complex to understand what's going on here

Reply:
agreed, got rid of that logic - now category search filters are applied directly in `useReceiptTableFilters` hook with all the other filters for search